File: src/test/java/social/alone/server/BaseRepositoryTest.java
package social.alone.server;

import social.alone.server.interest.Interest;
import social.alone.server.interest.InterestRepository;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import org.springframework.transaction.annotation.Transactional;

@RunWith(SpringRunner.class)
@DataJpaTest
@ActiveProfiles("test")
@Ignore
@Transactional
public class BaseRepositoryTest {

  @Autowired
  protected InterestRepository interestRepository;

  public Interest buildInterest(String value) {
    return interestRepository.save(
            new Interest(value)
    );
  }

}

File: src/test/kotlin/social/alone/server/BaseRepositoryTest.kt
package social.alone.server

import org.junit.Ignore
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.junit4.SpringRunner
import org.springframework.transaction.annotation.Transactional
import social.alone.server.interest.Interest
import social.alone.server.interest.InterestRepository

@RunWith(SpringRunner::class)
@DataJpaTest
@ActiveProfiles("test")
@Ignore
@Transactional
class BaseRepositoryTest {

    @Autowired
    lateinit protected var interestRepository: InterestRepository

    fun buildInterest(value: String): Interest {
        return interestRepository.save(
                Interest(value)
        )
    }

}
